src/Microsoft.DotNet.Watcher.Core/Internal/Implementation/Project.cs

```diff
@@ -15,12 +17,27 @@ public Project(ProjectModel.Project runtimeProject)
             ProjectFile = runtimeProject.ProjectFilePath;
             ProjectDirectory = runtimeProject.ProjectDirectory;
 
-            Files = runtimeProject.Files.SourceFiles.Concat(
-                    runtimeProject.Files.ResourceFiles.Values.Concat(
-                    runtimeProject.Files.PreprocessSourceFiles.Concat(
-                    runtimeProject.Files.SharedFiles))).Concat(
-                    new string[] { runtimeProject.ProjectFilePath })
-                .ToList();
+            var compilerOptions = runtimeProject.GetCompilerOptions(targetFramework: null, configurationName: null);
+
+            var filesToWatch = new List<string>() { runtimeProject.ProjectFilePath };
+            if (compilerOptions.CompileInclude != null)
+            {
+                filesToWatch.AddRange(compilerOptions.CompileInclude.ResolveFiles());
+
+                if (compilerOptions.EmbedInclude != null)
```

Comment:
In this case, if `CompileInclude` is not set but `EmbedInclude` is set it will be ignored.
This should probably be refactored to something like this.
```c#
if (compilerOptions.CompileInclude != null)
{
    filesToWatch.AddRange(compilerOptions.CompileInclude.ResolveFiles());
}
else
{
    filesToWatch.AddRange(runtimeProject.Files.SourceFiles);
}

if (compilerOptions.EmbedInclude != null)
{
    filesToWatch.AddRange(compilerOptions.EmbedInclude.ResolveFiles());
}
else
{
    filesToWatch.AddRange(runtimeProject.Files.ResourceFiles.Values);
}

filesToWatch.AddRange(runtimeProject.Files.PreprocessSourceFiles);
filesToWatch.AddRange(runtimeProject.Files.SharedFiles);
```